### PR TITLE
Added new folder for week-4 homework

### PR DESCRIPTION
**Pull request by Nusrath-Jahan**

### CODE REVIEW

File: JavaScript-1/JS-week-4/HomeWork/Voice Assistant/script.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+  // Handling the "What day is it today?" command
+  if (command === "What day is it today?") {
+    const today = new Date();
+    const options = { day: "numeric", month: "long", year: "numeric" };
+    return `Today is ${today.toLocaleDateString("en-US", options)}`;
+  }
+
+  // Handling simple math commands
+  const mathMatch = command.match(/What is (\d+) (\+|\-|\*|\/) (\d+)\?/);
```
Comment (vdaguenet) [inline JavaScript-1/JS-week-4/HomeWork/Voice Assistant/script.js:62]: Using a regex is an excellent idea!

**Review by vdaguenet — APPROVED** [review]
Very good homework! I am sure you can improve it even more by making small changes to the way you extract variables in commands.

File: JavaScript-1/JS-week-4/HomeWork/Voice Assistant/script.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    } else {
+      return "Your todo list is empty";
+    }
+  }
+
+  // Handling the "What day is it today?" command
+  if (command === "What day is it today?") {
+    const today = new Date();
+    const options = { day: "numeric", month: "long", year: "numeric" };
+    return `Today is ${today.toLocaleDateString("en-US", options)}`;
```
[the remark this answers is not in the document]
Comment (Nusrath-Jahan) [reply JavaScript-1/JS-week-4/HomeWork/Voice Assistant/script.js:58]: Thank you for your valuable comments.

File: JavaScript-1/JS-week-4/HomeWork/Voice Assistant/script.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
+let userName = "";
+let todos = [];
+
+function getReply(command) {
+  // Handling the "Hello my name is ..." command
+  if (command.startsWith("Hello my name is ")) {
+    const name = command.slice(17);
```
[the remark this answers is not in the document]
Comment (Nusrath-Jahan) [reply JavaScript-1/JS-week-4/HomeWork/Voice Assistant/script.js]: Hi vdaguenet ,

I have completed the tasks you assigned and pushed the updates to the branch. Please review the changes at your convenience.

Thank you

**Review by vdaguenet — APPROVED** [review]

File: JavaScript-1/JS-week-4/HomeWork/Voice Assistant/script.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (vdaguenet) [reply JavaScript-1/JS-week-4/HomeWork/Voice Assistant/script.js]: Excellent!

**Pull request merged**